engine/image-loader/image-loader.lisp

```diff
@@ -0,0 +1,47 @@
+(defpackage lbge.image-loader
+  (:use :cl)
+  (:export load-image make-image))
+(in-package lbge.image-loader)
+
+(defclass image ()
+  ((width :reader  width
+          :initarg :width
+          :initform nil
+          :type integer)
+   (height :reader  height
+           :initarg :height
+           :initform nil
+           :type integer)
+   (channels :reader  channels
+             :initarg :channels
+             :initform nil
+             :type string)
+   (data :reader  data
+         :initarg :data
+         :initform nil
+         :type string)))
+
+(defun make-image (&key width height channels data)
+  "Wrapper arround `image` class for making image objects
+   in subpackages."
+  (make-instance 'image
+                 :width width
+                 :height height
+                 :channels channels
+                 :data data))
+
+(defun load-image (path)
+  "Takes path to an image and returns `image` structure.
+   If image format unsupported throws error."
+  (cond-extension path
+    (".tga" (lbge.image-loader:tga path))
+    (t (error "Unsupported image format"))))
+
+(defmacro cond-extension (path &rest clouses)
```

Comment:
The macro can easily be replaced with built-in functionality. Not sure if it's needed at all in that case, but if so, it should probably use built-in functions anyway. Something like the following might suffice:
```common-lisp
;; Parse the path into a pathname and get the extension from that
;; ecase & ccase raise an error if the format is not in the list (ccase = correctable errors)
(ccase (pathname-type (parse-namestring path))
  ("tga" (do-the-tga-stuff))
  ("other-extension" (other-stuff)))
```